content/blog/page-view-counter/index.md

```diff
@@ -0,0 +1,273 @@
+     
+     table.Execute(TableOperation.InsertOrReplace(pageViewCount)); // insert or replace the entity with the new count
+     return (ActionResult)new OkObjectResult(pageViewCount.Count.ToString()); // return the view count 
+```
+
+Let's also make sure we return an error code if we don't give a "URL" identifier to the function before trying to retrieve an entity:
+
+```csharp
+  if (pageViewURL == null)
+  {
+    return (ActionResult)new StatusCodeResult(503);
+  }
+```
+The function is now ready! Let's make sure it works by running it locally and performing POST requests that have a "URL" parameter in the body.
+
+### Deploying and implementation
+
+We are now ready to deploy the Azure Function. Something that you might want to make sure of, is that you do not usually want to keep any connection string in code, so we will be getting it through the [App Settings](https://docs.microsoft.com/en-us/azure/azure-functions/functions-how-to-use-azure-function-app-settings#settings):
+
+```csharp
+  // we will have a StorageConnectionString field in the app settings where we store the storage account connection string
+  var storageAccountConnectionString = Environment.GetEnvironmentVariable("StorageConnectionString");
+```
+The App Settings can be changed in the Azure Portal at runtime.  
+
+Lastly, you can implement this on any page by creating a small script that calls into this function and gets the result:
+```js
+var url = 'https://pageviewcounterblog.azurewebsites.net/api/GetViewCounter/';
+var data = {URL: 'this-blog-post-URL'};
+
+fetch(url, {
+  method: 'POST', 
+  body: JSON.stringify(data), 
+  headers:{
+    'Content-Type': 'application/json'
+  }
+}).then(res => res.json())
+.then(response => console.log('Success:', JSON.stringify(response)))
+```
+
+Notice that this way, anyone that has access to the function endpoint can call into it. Securing the function goes beyond the scope of this post, especially because this is called in the frontend (whereas you would likely prefer to call this from the backend to authenticate properly and easily). However, something you can do pretty simply is to [add CORS policies](https://www.c-sharpcorner.com/article/handling-cors-in-azure-function/) to the function in order to have it be a little more restrictive.
+
+## Closing
+
+Right now you should have a working Azure Function that connects to Azure Storage Table and stores the amount of times that a certain page was visited. And this was all for free (except if you exceed the million requests per month, but you can probably handle paying at that point)! 
+
+This was meant to be a simple experiment that served as a good introduction to some Azure services, and to certain concepts related to computing in general. Plus, it shows a simple way in which you could extend your static sites through serverless computing.  
+
+Feel free to review the code here: https://github.com/IgnacioAmigo/PageViewCounterBlog.
```

Comment:
perfect! (dismiss previous comment about it)

Reply:
I still would love to see inside this blog post the entire Function code, so visitors can see all the things they just learned working together

Reply:
Also, it would be AWESOME to see this thing in action at least in this page, but if you want to skip it I would not hold it against you

Reply:
I thought about making it available for this page, I will work on that tomorrow.